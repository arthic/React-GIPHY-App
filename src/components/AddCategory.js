import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
	/* Deja el string vacio para que no te marque error ya que
	si no hay comillas recibe undefined*/
	const [inputValue, setInputValue] = useState('')
	const handleInputChange = (e) => {
		// console.log(e.target.value);
		setInputValue(e.target.value)
		console.log('Cambio el input');
	}
	const handleSubmit = (e) => {
		/*Comportamiento por defecto del formulario es la
		recarga de la pagina*/
		e.preventDefault()
		console.log('handleSubmit', inputValue);

		if (inputValue.trim().length > 2) {
			// Se llama como callback que tiene el estado anterior
			setCategories(cats => [inputValue, ...cats])
			setInputValue('')
		}
	}
	/* const handleFocusChange = (e) => {
		setInputValue(e.target.value)
		// console.log(valorDefault);
		// setInputValue(e.target.value='')
		console.log(e.target.value);
		if (e.target.value === 'Busca un GIF!!!') {
			setInputValue(e.target.value='')
		}
		console.log(e.target.value);
	}*/
	return (
		/* El fragment "<>" agrupa elementos HTML o JSX, al usar
		un Form ya se esta agrupando el elemento, por lo tanto ya
		no es necesario el fragment*/
		<form onSubmit={handleSubmit}>
			<div className="search-contend">
				<i className="fas fa-search"></i>
				<input
					type="text"
					value={inputValue}
					onChange={handleInputChange}
					// onFocus={handleFocusChange}
				/>
			</div>
		</form>
	)
}
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}
export default AddCategory
