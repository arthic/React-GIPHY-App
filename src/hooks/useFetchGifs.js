import {useState, useEffect} from 'react'
import { getGifs } from '../helpers/getGifs'

export const useFetchGifs = (category) => {
	const [state, setstate] = useState({
		data: [],
		loading: true
	})
	// Solo se dispara cuando cambia la categoria
	useEffect( () => {
		// Hace la peticion HTTP
		getGifs(category)
			.then(imgs => {
				setTimeout(() => {
					setstate({
						data: imgs,
						loading: false
					})
				}, 2000)
			})
	/* [] Lista(array) de dependencias, "al estar vacio
	solo se dispararÃ¡ una vez" */
	}, [category])
	return state
}
