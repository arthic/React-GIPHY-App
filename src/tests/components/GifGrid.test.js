import React from 'react'
import '@testing-library/jest-dom'
import { GifGrid } from '../../components/GifGrid'
import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
/* Fingir cualquier llamada a ese archivo y controlar lo que
la infor que va a responder */
jest.mock('../../hooks/useFetchGifs')

describe('Testing en <GifGrid />', () => {
	const category = 'Test-Categoria'

	test('Correcto despliege de <GifGrid />', () => {
		// El controlar el useFetchGifs
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		})
		const wrapper = shallow(<GifGrid category={category}/>)
		expect(wrapper).toMatchSnapshot()
	})
	test('Mostrar items cuando se cargan gifs del hook "useFetchGifs"', () => {
		const gifs = [{
				id: 12345,
				title: 'Cualquier title',
				url: 'https://localhost/anywhere/ramdom-gif.gif'
			},
			{
				id: 54321,
				title: 'Cualquier title',
				url: 'https://localhost/anywhere/ramdom-gif.gif'
			}
		]
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		})
		const wrapper = shallow(<GifGrid category={category}/>)
		// Que no exista el loading
		expect(wrapper.find('p').exists()).toBe(false)
		// Evaluar que el <GifGridItem /> exista
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

		expect(wrapper).toMatchSnapshot()
	})
});

