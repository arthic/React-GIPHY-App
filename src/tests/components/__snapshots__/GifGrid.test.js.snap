// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en <GifGrid /> Correcto despliege de <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    Test-Categoria
  </h3>
  <p
    className="loading animate__animated animate__flash"
  >
    <i
      className="fas fa-cloud-download-alt"
    />
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing en <GifGrid /> Mostrar items cuando se cargan gifs del hook "useFetchGifs" 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    Test-Categoria
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={12345}
      key="12345"
      title="Cualquier title"
      url="https://localhost/anywhere/ramdom-gif.gif"
    />
    <GifGridItem
      id={54321}
      key="54321"
      title="Cualquier title"
      url="https://localhost/anywhere/ramdom-gif.gif"
    />
  </div>
</Fragment>
`;
